tasks/managed/push-rpms-to-pulp/push-rpms-to-pulp.yaml

```diff
@@ -0,0 +1,316 @@
+        value: $(params.DEFAULT_EXCLUDES)
+      - name: DATA_DIR
+        value: $(params.dataDir)
+      - name: PULP_DOMAIN
+        value: $(params.PULP_DOMAIN)
+      - name: PULP_SECRET_NAME
+        value: $(params.PULP_SECRET_NAME)
+      - name: RESULTS_DIR_PATH
+        value: $(params.resultsDirPath)
+      - name: ociStorage
+        value: $(params.ociStorage)
+      - name: ociArtifactExpiresAfter
+        value: $(params.ociArtifactExpiresAfter)
+      - name: trustedArtifactsDebug
+        value: $(params.trustedArtifactsDebug)
+      - name: IMAGES_TXT
+        value: $(params.dataDir)/images.txt
+      - name: FILES_DIR
+        value: $(params.dataDir)/files
+      - name: IMAGE_EXPIRES_AFTER
+        value: $(params.ociArtifactExpiresAfter)
+      - name: ORAS_OPTIONS
+        value: $(params.orasOptions)
+      - name: DEBUG
+        value: $(params.trustedArtifactsDebug)
+    volumeMounts:
+      - name: workdir
+        mountPath: "/var/workdir"
+    workingDir: "/var/workdir"
+  steps:
+    - name: use-trusted-artifact
+      computeResources:
+        limits:
+          memory: 64Mi
+        requests:
+          memory: 64Mi
+          cpu: 30m
+      ref:
+        resolver: "git"
+        params:
+          - name: url
+            value: $(params.taskGitUrl)
+          - name: revision
+            value: $(params.taskGitRevision)
+          - name: pathInRepo
+            value: stepactions/use-trusted-artifact/use-trusted-artifact.yaml
+      params:
+        - name: workDir
+          value: $(params.dataDir)
+        - name: sourceDataArtifact
+          value: $(params.sourceDataArtifact)
+    - name: push-rpms-to-pulp
+      image: quay.io/konflux-ci/release-service-utils:d39d3af868c7136f7e4bffb32d1e625398d6264a
+      computeResources:
+        limits:
+          memory: 256Mi
+        requests:
+          memory: 256Mi
+          cpu: 250m
+      env:
+        - name: CONFIG_FILE_CONTENT
+          valueFrom:
+            secretKeyRef:
+              name: $(params.PULP_SECRET_NAME)
+              key: cli.toml
+      script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        set +x
+        echo "${CONFIG_FILE_CONTENT:?}" > "/tmp/cli.toml"
+        export PULP_CONFIG_FILE="/tmp/cli.toml"
+        set -x
+
+        DOMAIN_EXISTS=$(pulp --config "$PULP_CONFIG_FILE" domain show --name "${PULP_DOMAIN}" \
+          | jq -r ".name // \"\"" || echo "")
+        if [[ "$DOMAIN_EXISTS" == "$PULP_DOMAIN" ]]; then
+            echo "Domain '${PULP_DOMAIN}' exists"
+        else
+            echo "Domain '${PULP_DOMAIN}' not found."
+            exit 1
+        fi
+
+        RESULTS_FILE="${DATA_DIR}/${RESULTS_DIR_PATH}/push-rpms-to-pulp-results.json"
+        mkdir -p "$(dirname "${RESULTS_FILE}")"
+
+        # Initialize results file with empty rpmfiles array
+        echo '{"rpmfiles": []}' > "$RESULTS_FILE"
+
+        < "${SNAPSHOT_PATH}" jq -r '.components[].containerImage' | tee "${IMAGES_TXT}"
```

Comment:
what does this do?

Reply:
pipes `"${SNAPSHOT_PATH}"` into `jq` and then writes into `"${IMAGES_TXT}"`

Reply:
Yes. Want me to switch to the method you mentioned?

Reply:
I think it is clearer since it follows the style used in many other places, but up to you

Reply:
This got merged before I could address it. I'll follow up with another PR to retrofit

Reply:
thanks!